llvm/unittests/Analysis/MLModelRunnerTest.cpp

```diff
@@ -60,6 +65,64 @@ class MockAOTModel final {
     }
   }
 };
+
+class AdditionAOTModel final : public MockAOTModelBase {
+public:
+  AdditionAOTModel() = default;
+  void Run() override { R = A + B; }
+};
+
+class DiffAOTModel final : public MockAOTModelBase {
+public:
+  DiffAOTModel() = default;
+  void Run() override { R = A - B; }
+};
+
+static const char *M1Selector = "the model that subtracts";
+static const char *M2Selector = "the model that adds";
+
+static auto Hash1 = MD5::hash(arrayRefFromStringRef(M1Selector));
+static auto Hash2 = MD5::hash(arrayRefFromStringRef(M2Selector));
```

Comment:
Maybe use `MD5::update(StringRef)` here? Also, I think explicitly typing as `MD5Result` might be better as it isn't immediately obvious that's what `MD5::hash` returns.

Reply:
I'd rather not. It seems to imply statefulness. This makes it clear, imho, that computing this hash is all there is to do here, there's no follow-up.

Reply:
Sounds good re reasoning for not using `MD5::update` (after offline discussion).

Same comment regarding explicitly typing instead of `auto` though unless there's some compelling reason to not explicitly type as `MD5Result`.

Reply:
done.

Reply:
It doesn't appear to be fixed with the commit you just pushed?

Reply:
because I didn't push it

Reply:
Oh, yep. I was looking at the wrong commit.